<launch>

  <arg name="veh" default="husky1" />

  <group ns="$(arg veh)">
  <!-- Input arguments -->
    <arg name="device" value="6843" doc="TI mmWave sensor device type [1443,6843]" />
    <arg name="config" value="3d" doc="TI mmWave sensor device configuration" />
    <arg name="max_allowed_elevation_angle_deg" default="90" doc="Maximum allowed elevation angle in degrees for detected object data [0 > value >= 90]}" />
    <arg name="max_allowed_azimuth_angle_deg" default="90" doc="Maximum allowed azimuth angle in degrees for detected object data [0 > value >= 90]}" />
    <!-- <arg name="veh" default="husky1" /> -->
    <arg name="serial" default="00789E23" />

    <!-- mmWave_Manager node -->
    <node pkg="ti_mmwave_rospkg" type="ti_mmwave_rospkg" name="ti_mmwave" ns="radar_0" output="screen">
      <param name="command_port" value="/dev/mm_$(arg serial)_0" />
      <param name="command_rate" value="115200" />
      <param name="data_port" value="/dev/mm_$(arg serial)_1" />
      <param name="data_rate" value="921600" />
      <param name="max_allowed_elevation_angle_deg" value="$(arg max_allowed_elevation_angle_deg)" />
      <param name="max_allowed_azimuth_angle_deg" value="$(arg max_allowed_azimuth_angle_deg)" />
      <param name="frame_id" value="mmwave_right_link" />
      <!-- <remap from="/ti_mmwave/radar_scan_pcl" to="ti_mmwave/radar_scan_pcl" /> -->
    </node>

    <!-- mmWaveQuickConfig node (terminates after configuring mmWave sensor) -->
    <node pkg="ti_mmwave_rospkg" type="mmWaveQuickConfig" name="ti_mmwave_config" ns="radar_0" args="$(find ti_mmwave_rospkg)/cfg/IWR6843_AOP.cfg" output="screen" />
  
  </group>

</launch>
